ws-server/src/server.rs

```diff
@@ -139,49 +143,59 @@ impl<'a> Future for Incoming<'a> {
 	}
 }
 
-async fn handshake(
-	socket: tokio::net::TcpStream,
-	conn_id: ConnectionId,
-	methods: &Methods,
-	cfg: &Settings,
-	stop_monitor: &StopMonitor,
-) -> Result<(), Error> {
+enum Handshake<'a> {
+	Reject { status_code: u16 },
+	Accept { conn_id: ConnectionId, methods: &'a Methods, cfg: &'a Settings, stop_monitor: &'a StopMonitor },
+}
+
+async fn handshake(socket: tokio::net::TcpStream, details: Handshake<'_>) -> Result<(), Error> {
```

Comment:
I'm struggling a little with the readability of `details`, can we rename to `handshake_input`, `handshake` or something like that?

Reply:
Yeah, I'm open for suggestions for naming scheme here, I'm thinking maybe naming the enum `HandshakeKind`, so it would be `kind: HandshakeKind<'_>`. Alternatively `-Mode`?

Reply:
mode or kind is works :+1: